accounting/filters/ChtcScheddDSIGpuFilter.py

```diff
@@ -0,0 +1,256 @@
+
+import statistics as stats
+from pathlib import Path
+from .BaseFilter import BaseFilter
+
+
+DEFAULT_COLUMNS = {
+    15: "All GPU Hours",
+    20: "Num Uniq Job Ids",
+    240: "Max Rqst Gpus",
+}
+
+
+DEFAULT_FILTER_ATTRS = [
+    "RemoteWallClockTime",
+    "CommittedTime",
+    "RequestCpus",
+    "RequestGpus",
+    "RequestMemory",
+    "RecordTime",
+    "JobCurrentStartDate",
+    "MemoryUsage",
+    "NumJobStarts",
+    "NumShadowStarts",
+    "JobUniverse",
+    "JobStatus",
+    "EnteredCurrentStatus",
+    "BytesSent",
+    "BytesRecvd",
+    "LastRemoteHost"
+]
+
+
+class ChtcScheddDSIGpuFilter(BaseFilter):
+    name = "DSI GPU schedd job history"
+
+    def get_query(self, index, start_ts, end_ts, **kwargs):
+        # Returns dict matching Elasticsearch.search() kwargs
+        # (Dict has same structure as the REST API query language)
+        query = super().get_query(index, start_ts, end_ts, **kwargs)
+
+        query.update({
+            "body": {
+                "query": {
+                    "bool": {
+                        "filter": [
+                            {"range": {
+                                "RecordTime": {
+                                    "gte": start_ts,
+                                    "lt": end_ts,
+                                }
+                            }},
+                            {"range": {
+                                "RequestGpus": {
+                                    "gt": 0
+                                }
+                            }},
+                            {"term": {
+                                "JobStatus": 4
+                            }},
+                            {"regexp": {
+                                "LastRemoteHost.keyword": ".*dsigpu-?[0-9]+[.]chtc[.]wisc[.]edu"
+                            }},
+                        ]
+                    }
+                }
+            }
+        })
+        return query
+
+    def project_filter(self, data, doc):
+
+        # Get input dict
+        i = doc["_source"]
+
+        # Get output dict for this project
+        project = i.get("ProjectName", "UNKNOWN") or "UNKNOWN"
+        o = data["Projects"][project]
+
+        # Add custom attrs to the list of attrs
+        filter_attrs = DEFAULT_FILTER_ATTRS.copy()
+        filter_attrs = filter_attrs + ["User"]
+
+        # Count number of DAGNode Jobs
+        if i.get("DAGNodeName") is not None and i.get("JobUniverse")!=12:
```

Comment:
If Elasticsearch is already restricting the job universe...
```suggestion
        if i.get("DAGNodeName") is not None:
```